cloudify_cli/config/parser_config.py

```diff
@@ -166,6 +178,111 @@ def parser_config():
+                    },
+                    'download': {
+                        'arguments': {
+                            '-s,--snapshot-id': snapshot_id_argument(),
+                            '-o,--output': {
+                                'metavar': 'OUTPUT',
+                                'type': str,
+                                'help': 'The output file path of the snapshot to be downloaded',
+                                'dest': 'output',
+                                'required': False
+                            }
+                        },
+                        'help': 'command for downloading a snapshot from the management server',
+                        'handler': cfy.snapshots.download
+                    },
+                    'list': {
+                        'help': 'command for listing all snapshots on the '
+                                'Manager',
+                        'handler': cfy.snapshots.ls
+                    },
+                    'delete': {
+                        'arguments': {
+                            '-s,--snapshot-id': snapshot_id_argument()
+                        },
+                        'help': 'command for deleting a snapshot',
+                        'handler': cfy.snapshots.delete
+                    },
+                    'restore': {
+                        'arguments': {
+                            '-s,--snapshot-id': snapshot_id_argument(),
+                            '--without-deployments-envs': {
+                                'dest': 'without_deployments_envs',
+                                'action': 'store_true',
+                                'default': False,
+                                'help': 'Specify this flag to restore snapshot '
+                                        'without deployments environments'
+                            }
+                        },
+                        'help': 'command for restoring manager to specific snapshot',
+                        'handler': cfy.snapshots.restore
+                    }
+                }
+            },
+            'agents': {
+                'help': "Manages Cloudify's Agents",
+                'sub_commands': {
+                    'install': {
+                        'arguments': {
+                            '-d,--deployment-id': deployment_id_argument(
+                                hlp='The id of the deployment to install agents for, if not provided '
+                                'then command will install agents for all deployments'
+                            ),
+                            '-l,--include-logs': {
+                                'dest': 'include_logs',
+                                'action': 'store_true',
+                                'help': 'A flag whether to include logs in returned events'
+                            }
+                        },
+                        'help':'command for installing agents on deployments',
```

Comment:
If I understand correctly "host agents" then it is exactly what it does.

It installs new agents on application vms and makes sure those agents are connected to current manager.

It is a part of agent upgrade process.